web/src/main/java/com/epam/rft/atsy/web/controllers/NewApplicationPopupController.java

```diff
@@ -30,20 +41,43 @@ public ModelAndView loadPage() {
   }
 
   @RequestMapping(method = RequestMethod.POST, value = "/secure/new_application_popup")
-  public String saveOrUpdate(@Valid @ModelAttribute StateDTO stateDTO, BindingResult result) {
-    if (!result.hasErrors()) {
-      stateDTO.setStateType("newstate");
-      stateDTO.setStateIndex(0);
-
-      ApplicationDTO applicationDTO = ApplicationDTO.builder()
-          .creationDate(new Date())
-          .candidateId(stateDTO.getCandidateId())
-          .positionId(stateDTO.getPosition().getId())
-          .channelId(stateDTO.getChannel().getId())
-          .build();
-
-      applicationsService.saveApplicaton(applicationDTO, stateDTO);
+  public String saveOrUpdate(@Valid @ModelAttribute StateDTO stateDTO, BindingResult result)
+      throws ObjectNotFoundException {
+    if (result.hasErrors()) {
+      throw new IllegalArgumentException();
     }
+    validateIdsBeforeExecuteSaveOrUpdate(stateDTO.getCandidateId(), stateDTO.getChannel().getId(),
+        stateDTO.getPosition().getId());
+    stateDTO.setStateType("newstate");
+    stateDTO.setStateIndex(0);
+
+    ApplicationDTO applicationDTO = ApplicationDTO.builder()
+        .creationDate(new Date())
+        .candidateId(stateDTO.getCandidateId())
+        .positionId(stateDTO.getPosition().getId())
+        .channelId(stateDTO.getChannel().getId())
+        .build();
+
+    applicationsService.saveApplicaton(applicationDTO, stateDTO);
     return "redirect:/secure/candidate/" + stateDTO.getCandidateId();
+
+  }
+
+  private void validateIdsBeforeExecuteSaveOrUpdate(Long candidateId, Long channelId,
+                                                    Long positionId)
+      throws ObjectNotFoundException {
+    Assert.notNull(candidateId);
+    Assert.notNull(channelId);
+    Assert.notNull(positionId);
+
+    if (candidateService.getCandidate(candidateId) == null) {
```

Comment:
If you keep the getCandidate method in CandidateService, then you should throw the CandidateNotFoundException from the service.